File: HCWMPA&S/Practices/Practice2.2/single.cpp
#include <ctime>
#include <stdio.h>
#include <conio.h>

#define STEPS 10000000
#define RECORD_NUM 256

double mu = 0.5;
double x = 0;

double records[RECORD_NUM] = {0};

int main()
{
    clock_t start = clock();
    for (int i = 0; i < STEPS - RECORD_NUM; i++)
    {
        x = 1 - mu * x * x;
    }

    for (int i = 0; i < RECORD_NUM; i++)
    {
        x = 1 - mu * x * x;
        records[i] = x;
    }
    clock_t end = clock();
    double elapsed = double(end - start) / CLOCKS_PER_SEC;
    printf("Elapsed: %.3f seconds\n", elapsed);

    printf("Press enter to output the latest records ...\n");
    getch();

    for (int i = 0; i < RECORD_NUM; i++)
    {
        printf("%3d: %6.3lf\n", i, records[i]);
    }
    
    return 0;
}

File: HCWMPA&S/Practices/Practice2.1/single.py
import time

steps = 10000000
record_num = 256

mu = 0.5
x = 0.0

records = [0] * record_num

start = time.perf_counter()
for i in range(steps - record_num):
    x = 1 - mu * x * x
for i in range(record_num):
    x = 1 - mu * x * x
    records[i] = x
end = time.perf_counter()

print(f"Elapsed: {end - start:.4f} seconds")
input("Press enter to output the latest records ...")
for i in range(len(records)):
    print(f"{i + 1:3d}: {records[i]: .3f}")
